packages/babel-preset-expo/plugins/remap-react-native-imports.js

```diff
@@ -0,0 +1,279 @@
+  Linking: 'Libraries/Linking/Linking',
+  LogBox: 'Libraries/LogBox/LogBox',
+  NativeDialogManagerAndroid: [
+    'Libraries/NativeModules/specs/NativeDialogManagerAndroid',
+    'default',
+  ],
+  NativeEventEmitter: 'Libraries/EventEmitter/NativeEventEmitter',
+  Networking: 'Libraries/Network/RCTNetworking',
+  PanResponder: 'Libraries/Interaction/PanResponder',
+  PermissionsAndroid: 'Libraries/PermissionsAndroid/PermissionsAndroid',
+  PixelRatio: 'Libraries/Utilities/PixelRatio',
+  PushNotificationIOS: 'Libraries/PushNotificationIOS/PushNotificationIOS',
+  Settings: 'Libraries/Settings/Settings',
+  Share: 'Libraries/Share/Share',
+  StatusBarIOS: 'Libraries/Components/StatusBar/StatusBarIOS',
+  StyleSheet: 'Libraries/StyleSheet/StyleSheet',
+  Systrace: 'Libraries/Performance/Systrace',
+  ToastAndroid: 'Libraries/Components/ToastAndroid/ToastAndroid',
+  TurboModuleRegistry: 'Libraries/TurboModule/TurboModuleRegistry',
+  TVEventHandler: 'Libraries/Components/AppleTV/TVEventHandler',
+  UIManager: 'Libraries/ReactNative/UIManager',
+  unstable_batchedUpdates: ['Libraries/Renderer/shims/ReactNative', 'unstable_batchedUpdates'],
+  useColorScheme: ['Libraries/Utilities/useColorScheme', 'default'],
+  useWindowDimensions: ['Libraries/Utilities/useWindowDimensions', 'default'],
+  UTFSequence: 'Libraries/UTFSequence',
+  Vibration: 'Libraries/Vibration/Vibration',
+  YellowBox: 'Libraries/YellowBox/YellowBoxDeprecated',
+  DeviceEventEmitter: 'Libraries/EventEmitter/RCTDeviceEventEmitter',
+  NativeAppEventEmitter: 'Libraries/EventEmitter/RCTNativeAppEventEmitter',
+  NativeModules: 'Libraries/BatchedBridge/NativeModules',
+  Platform: 'Libraries/Utilities/Platform',
+  processColor: 'Libraries/StyleSheet/processColor',
+  PlatformColor: ['Libraries/StyleSheet/PlatformColorValueTypes', 'PlatformColor'],
+  DynamicColorIOS: ['Libraries/StyleSheet/PlatformColorValueTypesIOS', 'DynamicColorIOS'],
+  ColorAndroid: ['Libraries/StyleSheet/PlatformColorValueTypesAndroid', 'ColorAndroid'],
+  requireNativeComponent: 'Libraries/ReactNative/requireNativeComponent',
+  unstable_RootTagContext: 'Libraries/ReactNative/RootTagContext',
+  ColorPropType: 'Libraries/DeprecatedPropTypes/DeprecatedColorPropType',
+  EdgeInsetsPropType: 'Libraries/DeprecatedPropTypes/DeprecatedEdgeInsetsPropType',
+  PointPropType: 'Libraries/DeprecatedPropTypes/DeprecatedPointPropType',
+  ViewPropTypes: 'Libraries/DeprecatedPropTypes/DeprecatedViewPropTypes',
+};
+
+const pkg = 'react-native';
+
+const getDistLocation = importName => {
+  if (!importName || importName === 'index') {
+    // to prevent checking again and causing a loop
+    return 'react-native/index';
+  }
+
+  if (importName && importName in MAPPING) {
+    if (Array.isArray(MAPPING[importName])) {
+      return [`react-native/${MAPPING[importName][0]}`, MAPPING[importName][1]];
+    }
+    return `react-native/${MAPPING[importName]}`;
+  }
```

Comment:
* importName is defined here
* hasOwnProperty is safer if RN were to export something with the same name as a property on Object.prototype
* If RN adds a new module we want to learn about it instead of silently getting build time or file size regressions. This is especially important because RN is unstable and changed separately from this repo.


```suggestion
  if (MAPPING.hasOwnProperty(importName)) {
    if (Array.isArray(MAPPING[importName])) {
      return [`react-native/${MAPPING[importName][0]}`, MAPPING[importName][1]];
    }
    return `react-native/${MAPPING[importName]}`;
  } else {
    console.warn(`remap-react-native-imports cannot map { $importName } from react-native to its internal module. This may cause slower build times and larger file sizes.`);
  }
```